src/haxelib/client/Main.hx

```diff
@@ -611,8 +616,12 @@ class Main {
 		return password;
 	}
 
+	function getWritableRepository(reps: Array<String>):String {
+		return reps[0];
```

Comment:
Yes, ```haxelib update``` is also questionable, it may end up using the read-only repo where previous version is installed.